src/test/java/fi/fmi/avi/archiver/message/populator/ProductMessageTypesValidatorTest.java

```diff
@@ -1,91 +1,135 @@
     }
 
+    @Test
+    void valid_taf() {
+        final ArchiveAviationMessage.Builder builder = ArchiveAviationMessage.builder().setType(MessagePopulatorTests.TypeId.TAF.getId());
+        productMessageTypesValidator.populate(InputAviationMessage.builder().setFileMetadata(TAF_METADATA).buildPartial(), builder);
+        assertThat(builder.getProcessingResult()).isEqualTo(ProcessingResult.OK);
+    }
+
     @Test
     void invalid_type() {
         final ArchiveAviationMessage.Builder builder = ArchiveAviationMessage.builder().setType(-1);
-        productMessageTypesValidator.populate(InputAviationMessage.builder().setFileMetadata(TEST_METADATA).buildPartial(), builder);
+        productMessageTypesValidator.populate(InputAviationMessage.builder().setFileMetadata(METAR_METADATA).buildPartial(), builder);
         assertThat(builder.getProcessingResult()).isEqualTo(ProcessingResult.FORBIDDEN_MESSAGE_TYPE);
     }
 
+    @Test
+    void misconfigured_message_type() {
+        assertThrows(IllegalArgumentException.class, () ->
+                new ProductMessageTypesValidator(MessagePopulatorTests.TYPE_IDS, PRODUCTS, ImmutableMap.of(TAF_PRODUCT, Collections.singleton(new MessageType("test")))));
+
+    }
+
+    @Test
+    void misconfigured_product() {
+        assertThrows(IllegalArgumentException.class, () ->
+                new ProductMessageTypesValidator(MessagePopulatorTests.TYPE_IDS, PRODUCTS, ImmutableMap.of(OTHER_PRODUCT, Collections.singleton(MessageType.METAR))));
+    }
+
+    // Test nulls apart from constructor
     @Test
     public void testNulls() {
         final Class<?> classUnderTest = ProductMessageTypesValidator.class;
         final NullPointerTester tester = new NullPointerTester();
         final NullPointerTester.Visibility minimalVisibility = NullPointerTester.Visibility.PACKAGE;
         tester.setDefault(ArchiveAviationMessage.Builder.class, ArchiveAviationMessage.builder());
-        tester.setDefault(String.class, "test");
-        tester.setDefault(Map.class, MessagePopulatorTests.TYPE_IDS);
         tester.setDefault(InputAviationMessage.class, InputAviationMessage.builder().buildPartial());
 
         tester.testStaticMethods(classUnderTest, minimalVisibility);
-        tester.testConstructors(classUnderTest, minimalVisibility);
         tester.testInstanceMethods(productMessageTypesValidator, minimalVisibility);
     }
 
+    // Test constructor nulls
+    @Test
+    void testConstructorNulls() {
+        assertThrows(NullPointerException.class, () -> new ProductMessageTypesValidator(null, PRODUCTS, PRODUCT_MESSAGE_TYPES));
+        assertThrows(NullPointerException.class, () -> new ProductMessageTypesValidator(MessagePopulatorTests.TYPE_IDS, null, PRODUCT_MESSAGE_TYPES));
+        assertThrows(NullPointerException.class, () -> new ProductMessageTypesValidator(MessagePopulatorTests.TYPE_IDS, PRODUCTS, null));
```

Comment:
For consistency, I would suggest using AssertJ assertion.

```suggestion
        assertThatNullPointerException().isThrownBy(() -> new ProductMessageTypesValidator(null, PRODUCTS, PRODUCT_MESSAGE_TYPES));
        assertThatNullPointerException().isThrownBy(() -> new ProductMessageTypesValidator(MessagePopulatorTests.TYPE_IDS, null, PRODUCT_MESSAGE_TYPES));
        assertThatNullPointerException().isThrownBy(() -> new ProductMessageTypesValidator(MessagePopulatorTests.TYPE_IDS, PRODUCTS, null));
```